clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -3332,6 +3388,39 @@ checkBuiltinTemplateIdType(Sema &SemaRef, BuiltinTemplateDecl *BTD,
     }
     return HasNoTypeMember;
   }
+
+  case BTK__hlsl_spirv_type: {
+    assert(Converted.size() == 4);
+
+    if (!Context.getTargetInfo().getTriple().isSPIRV()) {
+      SemaRef.Diag(TemplateLoc, diag::err_hlsl_spirv_only)
+          << "__hlsl_spirv_type";
```

Comment:
Can this instead be?
```suggestion
          << BTD;
```
If not, we should put this string into the error message directly. We tend to avoid hard coded strings as inputs to diagnostics.

Reply:
Done.